salt/states/boto_ec2.py

```diff
@@ -1752,5 +1749,169 @@ def private_ips_absent(name, network_interface_name=None, network_interface_id=N
 
         # there were no changes since we did not attempt to remove ips
         ret['changes'] = {}
+    return ret
+
+
+def secgroup_present(
```

Comment:
A use-case I'm encountering is that I want to change the security-groups after the instance has already been created (by salt). For instance, the default salt bootstrap requires access to Internet, but some EC2 instances should not have Internet access at all. In order to fix this, I create the instance with the relevant securitygroup, but update the security-groups in a later stage to not have the security-group that provides Internet Access. A later stage does not need to have (nor has) all the other EC2 instance configuration data, so a full `boto_ec2.instance_present` with all the same attributes as the original creation call, except the Internet access security group, is not feasible.

Another use-case is where I temporarily want to have a security-group present during a state run for temporary access from the Internet, and remove that security-group again later. You really don't want to spam the entire `boto_ec2.instance_present` with all its creation parameters twice in a state file (at least, I don't) just for a temporary security-group modification.